planner/core/rule_column_pruning.go

```diff
@@ -117,6 +119,17 @@ func (la *LogicalAggregation) PruneColumns(parentUsedCols []*expression.Column)
 	for _, aggrFunc := range la.AggFuncs {
 		selfUsedCols = expression.ExtractColumnsFromExpressions(selfUsedCols, aggrFunc.Args, nil)
 	}
+	if len(la.GroupByItems) == 0 && len(la.AggFuncs) == 0 {
```

Comment:
If `len(la.GroupByItems) > 0 && len(la.AggFuncs) == 0`, we'd better add this dummy aggregation function as well. Though the result is correct now, we are depending on the specific implementation of executor, if executor is changed in the future, we may have problem for this case then.

Reply:
Thanks for the explaination